torch/utils/viz/MemoryViz.js

```diff
@@ -1271,6 +1271,36 @@ function create_trace_view(
   plot.set_delegate(delegate);
 }
 
+function create_settings_view(dst, snapshot, device) {
```

Comment:
To make this easy to maintain maybe just do:
`settings_div.append('pre').text(JSON.stringify(snapshot.allocator_settings))`. It will handle whatever nested stuff we put in there without having to update this page.

Reply:
@zdevito I tried this out, but it doesn't not look very good. Everything is put into one line.
![image](https://github.com/user-attachments/assets/9dbdd2f9-23aa-4a02-98a0-011350bb4a51)

Reply:
I've discovered JSON pretty printing, and it looks so much better, let's go with that one:
![image](https://github.com/user-attachments/assets/8b3df71c-f987-4604-b4b9-7b2eb7d4a75c)